src/pudl_archiver/archivers/ferc/xbrl.py

```diff
@@ -312,6 +354,15 @@ async def archive_year(
             filename = f"{filing.title}_form{filing.ferc_formname.as_int()}_{filing.ferc_period}_{round(filing.published_parsed.timestamp())}.xbrl".replace(
                 " ", "_"
             )
+            filing_name = f"{filing.title}{filing.ferc_period}"
+            if filing_name not in metadata:
+                metadata[filing_name] = []
+
+            filing_metadata = FilingMetadata.from_rss_metadata(
```

Comment:
If you remove `filename` as a parameter we can also remove lines 354-356!

Reply:
Removed by using `defaultdict`!